pkg/asset/tls/tls.go

```diff
@@ -196,3 +190,30 @@ func GenerateSelfSignedCertificate(cfg *CertCfg) (*rsa.PrivateKey, *x509.Certifi
 	}
 	return key, crt, nil
 }
+
+// hasShortCertRotationEnabled returns true if ShortCertRotation featuregate is enabled.
+func hasShortCertRotationEnabled(installConfig *installconfig.InstallConfig) bool {
+	fgs := installConfig.Config.EnabledFeatureGates()
+	return fgs.Enabled(features.FeatureShortCertRotation)
+}
+
+// ValidityOneDay sets the validity of a cert to 24 hours - or 1 hour when ShortRotationEnabled featuregate is enabled.
+func ValidityOneDay(installConfig *installconfig.InstallConfig) time.Duration {
+	if hasShortCertRotationEnabled(installConfig) {
+		return time.Hour
+	}
+	return time.Hour * 24
+}
+
+// ValidityOneYear sets the validity of a cert to 1 year - or two hours when ShortRotationEnabled featuregate is enabled.
+func ValidityOneYear(installConfig *installconfig.InstallConfig) time.Duration {
+	if hasShortCertRotationEnabled(installConfig) {
+		return time.Hour * 2
```

Comment:
curious: Is this value chosen arbitrarily with some "short" duration? Or is there a specification for each component that uses this feature gate?

Reply:
Its arbitrary for now, but maybe have to change later.
The lower limit here is 30 mins - usually shorter certificates cause kube-apiserver disruption. 
The upper limit is 6 hours - so that e2e test could observe this certificate being updated